### PR TITLE
bt-333:  Forgot password functionality.

### LINKED ISSUE #333: Forgot password functionality
Need to implement forgot password flow (Screenshot 1)
- [x] describe the flow
- [x] create email package for example using SendGrid
- [x] made changes in db and add methods to services
- [x] create resetPassword endpoin

Little hint: (https://blog.logrocket.com/implementing-secure-password-reset-node-js/)

Screenshot 1
Image

### PR DESCRIPTION
**Pull request by Simpsonkk**
You should refresh your .env with new data from .env.example in the backend folder. 
MAIL_API_KEY should be filled with API KEY, I will send it in a private message in slack, write me please.

### CODE REVIEW

File: backend/src/bundles/email/email.service.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import sgMail from '@sendgrid/mail';
+
+import { config } from '~/common/packages/packages.js';
+
+class EmailService {
+    public constructor() {
+        sgMail.setApiKey(config.ENV.SEND_GRID.MAIL_API_KEY);
```
Comment (MaksymGavatuk) [inline backend/src/bundles/email/email.service.ts]: All variables from config provide to constructor and save to private fields

File: backend/src/bundles/email/email.service.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    }
+
+    public async sendForgotPasswordEmail(
+        email: string,
+        resetToken: string,
+    ): Promise<void> {
+        const link = `${config.ENV.SEND_GRID.CLIENT_URL}/reset-password/${resetToken}`;
```
Comment (MaksymGavatuk) [inline backend/src/bundles/email/email.service.ts]: The same

File: backend/src/bundles/email/email.service.ts [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        const message = {
+            to: email,
+            from: config.ENV.SEND_GRID.MAIL_SENDER_DOMAIN,
```
Comment (MaksymGavatuk) [inline backend/src/bundles/email/email.service.ts]: The same

File: backend/.env.example [hunk 4]
```diff
@@ -35,3 +35,10 @@ PASSWORD_SALT_ROUNDS=10
 AWS_ACCESS_KEY_ID=string
 AWS_SECRET_ACCESS_KEY=string
 AWS_BUCKET_NAME=string
+
+#
+# SENDGRID
+#
+MAIL_API_KEY=string
+MAIL_SENDER_DOMAIN=bsatalents@gmail.com
+CLIENT_URL=http://localhost:3000
```
Comment (MaksymGavatuk) [inline backend/.env.example]: EOF

**Review by MaksymGavatuk — CHANGES_REQUESTED** [review]

File: backend/src/bundles/email/email.service.ts (hunk 1 above)
Comment (Simpsonkk) [reply backend/src/bundles/email/email.service.ts]: done

File: backend/src/bundles/email/email.service.ts (hunk 2 above)
Comment (Simpsonkk) [reply backend/src/bundles/email/email.service.ts]: done

File: backend/.env.example (hunk 4 above)
Comment (Simpsonkk) [reply backend/.env.example]: done

File: backend/src/bundles/email/email.service.ts (hunk 3 above)
Comment (Simpsonkk) [reply backend/src/bundles/email/email.service.ts]: done

**Review by MaksymGavatuk — APPROVED** [review]
Looks good as for me

**Review by anton-otroshchenko — APPROVED** [review]

**Review by nikita-remeslov — APPROVED** [review]

**Comment by nikita-remeslov** [comment]
Conflicts

**Pull request merged**